### PR TITLE
Automated testing

### PR DESCRIPTION
**Pull request by xMestas**
Fixes bug in the PFA method, and adds integration tests for DRGEPSA and PFA

### CODE REVIEW

**Comment by xMestas** [comment]
The files expected to be produced by the reduction were generated and placed in the test directory.  For each method, the test will execute the reduction to produce a reduced solution, and read in the expected solution from the expected file.  Attributes of the final solution and the expected final solution are then compared for equivalence.

**Comment by xMestas** [comment]
Or the files generated by the test could just be deleted, since they aren't needed.  Would that work?

**Comment by kyleniemeyer** [comment]
Personally I think it's neater (and safer) to create a system temporary folder that will automatically be removed, rather than actually delete files—less room accidental deletion and leaving things behind.

**Comment by xMestas** [comment]
The problem is that the function being called in the test generates the file in the cwd.  Should I also make a change where it generates the file on a given path?  And then change the test to make a temp directory and then generate the file in the temp directory?

**Comment by kyleniemeyer** [comment]
Yes, I think the path should be changeable (as in, should be a function argument). And then have the test make a temp directory and make the file there.

**Comment by xMestas** [comment]
What about moving the function call that writes the file up to the pymars function?  No need to call it in every method when it can just be called once at the end.  This will stop the files from being generated by the test too, removing the need for a temp directory.  The path to which the file is written can still be added as an argument to just the pymars function and not the function that does the actual reduction.

**Comment by xMestas** [comment]
Pushed the changes stated above to the pull request

**Comment by kyleniemeyer** [comment]
Moving that function up seems fine, but regardless I think any creation of files should be done in one of these temporary folders—it's just cleaner and safer than anything else.

**Comment by xMestas** [comment]
Yup, the output files are no longer created by the test though, and those files are meant to be given to the user, so they shouldn't be deleted.  The hdf5 files probably should be stored in a tmp dir and deleted after, and I have that recorded as an issue.

**Comment by kyleniemeyer** [comment]
Looks like there is a merge conflict now?

**Comment by xMestas** [comment]
Merge conflict should be resolved now.

File: pymars/tests/test_methods.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+import drgep
+import pfa
+import sensitivity_analysis
+
+
+def testDRGEPSA():
+
+	# Original model
+	path_to_original = os.path.dirname(os.path.abspath(__file__)) + "/../../example_files/gri30.cti"
```
Comment (kyleniemeyer) [inline pymars/tests/test_methods.py]: this is fairly fragile, and I think there's a better way using `pkg_resources.resource_ filename()`.

In this case, something along the lines of
```
    def relative_location(self, file):
        file_path = os.path.join("example_files", file)
        return pkg_resources.resource_filename(__name__, file_path)
```
might work

Comment (xMestas) [reply pymars/tests/test_methods.py]: Had trouble getting this to work, but I tried something else and I think it is a little cleaner now.  See latest commit.

**Pull request merged**